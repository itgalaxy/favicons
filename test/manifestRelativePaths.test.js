const favicons = require("../src");
const { logo_png } = require("./util");

test("should images without options.path to manifests when manifestRelativePaths is true", async () => {
  expect.assertions(1);

  const result = await favicons(logo_png, {
    manifestRelativePaths: true,
    path: "favicons/",
    icons: {
      android: true,
      firefox: true,
      windows: true,
      yandex: true,
      appleIcon: false,
      appleStartup: false,
      coast: false,
      favicons: false,
    },
  });

  await expect(result).toMatchFaviconsSnapshot();
});
