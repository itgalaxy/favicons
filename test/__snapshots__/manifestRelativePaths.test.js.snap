// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should images without options.path to manifests when manifestRelativePaths is true 16`] = `
{
  "files": [
    {
      "contents": "{
  "name": null,
  "short_name": null,
  "description": null,
  "dir": "auto",
  "lang": "en-US",
  "display": "standalone",
  "orientation": "any",
  "start_url": "/?homescreen=1",
  "background_color": "#fff",
  "theme_color": "#fff",
  "icons": [
    {
      "src": "android-chrome-36x36.png",
      "sizes": "36x36",
      "type": "image/png",
      "purpose": "any"
    },
    {
      "src": "android-chrome-48x48.png",
      "sizes": "48x48",
      "type": "image/png",
      "purpose": "any"
    },
    {
      "src": "android-chrome-72x72.png",
      "sizes": "72x72",
      "type": "image/png",
      "purpose": "any"
    },
    {
      "src": "android-chrome-96x96.png",
      "sizes": "96x96",
      "type": "image/png",
      "purpose": "any"
    },
    {
      "src": "android-chrome-144x144.png",
      "sizes": "144x144",
      "type": "image/png",
      "purpose": "any"
    },
    {
      "src": "android-chrome-192x192.png",
      "sizes": "192x192",
      "type": "image/png",
      "purpose": "any"
    },
    {
      "src": "android-chrome-256x256.png",
      "sizes": "256x256",
      "type": "image/png",
      "purpose": "any"
    },
    {
      "src": "android-chrome-384x384.png",
      "sizes": "384x384",
      "type": "image/png",
      "purpose": "any"
    },
    {
      "src": "android-chrome-512x512.png",
      "sizes": "512x512",
      "type": "image/png",
      "purpose": "any"
    }
  ]
}",
      "name": "manifest.webmanifest",
    },
    {
      "contents": "<?xml version="1.0" encoding="utf-8"?>
<browserconfig>
  <msapplication>
    <tile>
      <square70x70logo src="mstile-70x70.png"/>
      <square150x150logo src="mstile-150x150.png"/>
      <wide310x150logo src="mstile-310x150.png"/>
      <square310x310logo src="mstile-310x310.png"/>
      <TileColor>#fff</TileColor>
    </tile>
  </msapplication>
</browserconfig>",
      "name": "browserconfig.xml",
    },
    {
      "contents": "{
  "version": "1.0",
  "api_version": 1,
  "layout": {
    "logo": "yandex-browser-50x50.png",
    "color": "#fff",
    "show_title": true
  }
}",
      "name": "yandex-browser-manifest.json",
    },
  ],
  "html": [
    "<link rel="manifest" href="/favicons/manifest.webmanifest">",
    "<meta name="mobile-web-app-capable" content="yes">",
    "<meta name="theme-color" content="#fff">",
    "<meta name="application-name">",
    "<meta name="msapplication-TileColor" content="#fff">",
    "<meta name="msapplication-TileImage" content="/favicons/mstile-144x144.png">",
    "<meta name="msapplication-config" content="/favicons/browserconfig.xml">",
    "<link rel="yandex-tableau-widget" href="/favicons/yandex-browser-manifest.json">",
  ],
  "images": [
    {
      "contents": null,
      "name": "android-chrome-36x36.png",
    },
    {
      "contents": null,
      "name": "android-chrome-48x48.png",
    },
    {
      "contents": null,
      "name": "android-chrome-72x72.png",
    },
    {
      "contents": null,
      "name": "android-chrome-96x96.png",
    },
    {
      "contents": null,
      "name": "android-chrome-144x144.png",
    },
    {
      "contents": null,
      "name": "android-chrome-192x192.png",
    },
    {
      "contents": null,
      "name": "android-chrome-256x256.png",
    },
    {
      "contents": null,
      "name": "android-chrome-384x384.png",
    },
    {
      "contents": null,
      "name": "android-chrome-512x512.png",
    },
    {
      "contents": null,
      "name": "mstile-70x70.png",
    },
    {
      "contents": null,
      "name": "mstile-144x144.png",
    },
    {
      "contents": null,
      "name": "mstile-150x150.png",
    },
    {
      "contents": null,
      "name": "mstile-310x150.png",
    },
    {
      "contents": null,
      "name": "mstile-310x310.png",
    },
    {
      "contents": null,
      "name": "yandex-browser-50x50.png",
    },
  ],
}
`;
